src/dualize.jl

```diff
@@ -6,42 +6,48 @@ export dualize
 Dualize the model
 """
 function dualize(primal_model::MOI.ModelLike; dual_names::DualNames = DualNames("", ""))
-    return dualize(primal_model, dual_names, Float64)
+    # Crates an empty dual problem
+    dual_problem = DualProblem{Float64}()
+    return dualize(primal_model, dual_problem, dual_names)
 end
 
-function dualize(primal_model::MOI.ModelLike, dual_names::DualNames, T::DataType)
+function dualize(primal_model::MOI.ModelLike, dual_problem::DualProblem{T}; 
+                 dual_names::DualNames = DualNames("", "")) where T
+    # Dualize with the optimizer already attached
+    return dualize(primal_model, dual_problem, dual_names)
+end
+
+function dualize(primal_model::MOI.ModelLike, dual_problem::DualProblem{T}, dual_names::DualNames) where T
```

Comment:
You don't seem to be using `T` in the function
```suggestion
function dualize(primal_model::MOI.ModelLike, dual_problem::DualProblem, dual_names::DualNames)
```